### ISSUE TITLE
Back button on /detail/{id} does nothing when opening a page in /details/{id}

### ISSUE THREAD

**Issue by mickeyvip** [johnpapa/angular2-tour-of-heroes#36]
As I pointed out in my comment to issue #33, if a user opens a page with link to `/details/{id}`, clicking the `back` button does nothing, because there is no history in the browser.

Thank you.

**Comment by ColinEberhardt** [comment]
I think this is as expected. Try the same with Google Mail, navigate to an email then open  anew browser window using the deep-link to the given email. If you click the back button you will find that it does not navigate back to the list of emails.

**Comment by mickeyvip** [comment]
Just tried it. `back` button in the menu brought me to the emails list, as I would expect.

![image](https://cloud.githubusercontent.com/assets/220427/11977415/4dc90572-a98b-11e5-9efe-68cfba5b042d.png)

**Comment by ColinEberhardt** [comment]
That's different, that button is within the mail application itself. Try the browsers back button instead.

I agree, this might not be what everyone expects, but it is a behaviour that most single page apps have settled on.

**Comment by johnpapa** [comment]
The back button is there to show how to go back and uses history. Of course we could add logic to see where is "back" going and divert it elsewhere. Or instead of back we could say "go to heroes". It's a tutorial on features, not a prod app tho :)

so while these are valid concerns, I think it's best to leave as is

**Comment by mickeyvip** [comment]
@ColinEberhardt, I am talking about `back` button in the application, not the browser's button.

@johnpapa, we can redirect the users where we want, as you say, but right now clicking the button does nothing and it's confusing and making the app feel buggy.

And as a tutorial I would love to see how we can navigate to the route we came from (or to default route if we came from "nowhere", by navigating to the specific page following a link). It would show some more router internals, which I could not figure out from the docs.

It's my personal opinion, of course :)
